pkg/globalnet/controllers/ingress_endpoints_controller.go

```diff
@@ -0,0 +1,153 @@
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/submariner-io/admiral/pkg/federate"
+	"github.com/submariner-io/admiral/pkg/stringset"
+	"github.com/submariner-io/admiral/pkg/syncer"
+	"github.com/submariner-io/admiral/pkg/util"
+	submarinerv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+)
+
+func startIngressEndpointsController(svc *corev1.Service, config syncer.ResourceSyncerConfig) (*ingressEndpointsController, error) {
+	var err error
+
+	_, gvr, err := util.ToUnstructuredResource(&submarinerv1.GlobalIngressIP{}, config.RestMapper)
+	if err != nil {
+		return nil, err
+	}
+
+	controller := &ingressEndpointsController{
+		baseSyncerController: newBaseSyncerController(),
+		svcName:              svc.Name,
+		namespace:            svc.Namespace,
+		ingressIPMap:         stringset.NewSynchronized(),
+	}
+
+	fieldSelector := fields.Set(map[string]string{"metadata.name": svc.Name}).AsSelector().String()
+
+	controller.resourceSyncer, err = syncer.NewResourceSyncer(&syncer.ResourceSyncerConfig{
+		Name:                "Ingress Endpoints syncer",
+		ResourceType:        &corev1.Endpoints{},
+		SourceClient:        config.SourceClient,
+		SourceNamespace:     svc.Namespace,
+		RestMapper:          config.RestMapper,
+		Federator:           federate.NewCreateFederator(config.SourceClient, config.RestMapper, corev1.NamespaceAll),
+		Scheme:              config.Scheme,
+		Transform:           controller.process,
+		SourceFieldSelector: fieldSelector,
+		ResourcesEquivalent: areEndpointsEqual,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if err := controller.Start(); err != nil {
+		return nil, err
+	}
+
+	ingressIPs := config.SourceClient.Resource(*gvr).Namespace(corev1.NamespaceAll)
+	controller.reconcile(ingressIPs, func(obj *unstructured.Unstructured) runtime.Object {
+		// TODO: Consider adding endpointsRef field to GlobalIngressIPSpec
+		endpointsName, exists, _ := unstructured.NestedString(obj.Object, "spec", "podRef", "name")
+		if exists {
+			return &corev1.Endpoints{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      endpointsName,
+					Namespace: obj.GetNamespace(),
+				},
+			}
+		}
+
+		return nil
+	})
+
+	klog.Infof("Created Endpoints controller for (%s/%s) with selector %q", svc.Namespace, svc.Name, fieldSelector)
+
+	return controller, nil
+}
+
+func (c *ingressEndpointsController) process(from runtime.Object, numRequeues int, op syncer.Operation) (runtime.Object, bool) {
+	endpoints := from.(*corev1.Endpoints)
+	key, _ := cache.MetaNamespaceKeyFunc(endpoints)
+
+	ingressIP := &submarinerv1.GlobalIngressIP{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("ep-%.59s", endpoints.Name),
```

Comment:
```suggestion
			Name:      fmt.Sprintf("ep-%.60s", endpoints.Name),
```

I think we can go with 60 here.

Reply:
Fixed.